### PR TITLE
[WebConsole] Add code conflict resolution on upstream updates

### PR DESCRIPTION
**Pull request by Karakatiza666**
Now if pipeline code is changed outside the code editor:
-  if the user has started editing the program they will be presented with conflict resolution options
- if the user hasn't edited the program the updated version will be displayed automatically
image

Fix https://github.com/feldera/feldera/issues/2621 : program text obscured by navigation text
Add code minimap toggle shortcut - Ctrl/Command+M

Fix https://github.com/feldera/feldera/issues/2696 : Web Console does not show all Rust errors
Now if no errors are recognized as relating to a position in a user-facing source file, all errors are displayed.

Fix https://github.com/feldera/feldera/issues/2684 : attempting to insert code into read-only section gives transparent warning

### CODE REVIEW

**Comment by Karakatiza666** [comment]
Updated error list derivation to show all errors reported by Rust, but show errors with known source position first in the list. Condition for displaying internal error message hasn't changed - if the stderr contains error in main.rs file

**Comment by Karakatiza666** [comment]
> creating a pipeline that got deleted re-adds it with the old SQL/stub
There is another issue related to this one, I will handle both in a separate PR

**Comment by gz** [comment]
it looks like there is an issue that the change detection doesn't pick it up when you modify the program twice in a row with the same code from external

Uploading Screencast from 2024-10-10 09-30-21.mp4…

**Review by gz — APPROVED** [review]

File: web-console/src/lib/compositions/health/systemErrors.ts [hunk 1]
```diff
@@ -183,16 +193,20 @@ export const extractProgramErrors =
       .returnType<SystemError<any, Report>[]>()
       .with({ RustError: P.any }, (e) => {
         const rustCompilerErrorRegex =
-          /((warning:|error:|error\[[\w]+\]:)[\s\S]+?)\n(\n|(?=( +Compiling|warning:|error:|error\[[\w]+\]:)))/g
+          /((warning:|^error:|error\[[\w]+\]:)[\s\S]+?)\n(\n|(?=( +Compiling|warning:|error:|error\[[\w]+\]:)))/gm
```
Comment (gz) [inline web-console/src/lib/compositions/health/systemErrors.ts:196]: looks scary.. fwiw there is an option that the rustc compiler just gives out JSON formatted error messages. maybe we should send you that @snkas @ryzhyk

**Review by gz — APPROVED** [review]

File: web-console/src/lib/compositions/health/systemErrors.ts (hunk 1 above)
Comment (Karakatiza666) [reply web-console/src/lib/compositions/health/systemErrors.ts:196]: Had to use regex until this is implemented

**Comment by Karakatiza666** [comment]
> it looks like there is an issue that the change detection doesn't pick it up

Discussed with Gerd, not actually an issue

**Pull request merged**